src/NPCManager.cpp

```diff
@@ -91,6 +91,119 @@ void NPCManager::init() {
     REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_WARP_USE_NPC, npcWarpHandler);
     REGISTER_SHARD_PACKET(P_CL2FE_REQ_NPC_SUMMON, npcSummonHandler);
     REGISTER_SHARD_PACKET(P_CL2FE_REQ_BARKER, npcBarkHandler);
+    REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_VENDOR_START, npcVendorStart);
+    REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_VENDOR_TABLE_UPDATE, npcVendorTable);
+    REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_VENDOR_ITEM_BUY, npcVendorBuy);
+}
+
+void NPCManager::npcVendorBuy(CNSocket* sock, CNPacketData* data) {
+    if (data->size != sizeof(sP_CL2FE_REQ_PC_VENDOR_ITEM_BUY))
+        return; // malformed packet
+
+    sP_CL2FE_REQ_PC_VENDOR_ITEM_BUY* req = (sP_CL2FE_REQ_PC_VENDOR_ITEM_BUY*)data->buf;
+    PlayerView& plr = PlayerManager::players[sock];
+
+    int itemCost = 100; //TODO: placeholder, look up the price of item
+
+    //TODO: need to check if user has enough money + free inventory space, else send fail packet 
+
+    if (itemCost > plr.plr->money) {
+        INITSTRUCT(sP_FE2CL_REP_PC_VENDOR_ITEM_BUY_FAIL, failResp);
+        failResp.iErrorCode = 0;
+        sock->sendPacket((void*)&failResp, P_FE2CL_REP_PC_VENDOR_ITEM_BUY_FAIL, sizeof(sP_FE2CL_REP_PC_VENDOR_ITEM_BUY_FAIL));
+        return;
+    }
+
+    INITSTRUCT(sP_FE2CL_REP_PC_VENDOR_ITEM_BUY_SUCC, resp);
+
+    plr.plr->money = plr.plr->money - itemCost;
+    plr.plr->Inven[resp.iInvenSlotNum] = req->Item;
+
+    resp.iCandy = plr.plr->money;
+    resp.iInvenSlotNum = req->iInvenSlotNum;
+    resp.Item = req->Item;
+
+    sock->sendPacket((void*)&resp, P_FE2CL_REP_PC_VENDOR_ITEM_BUY_SUCC, sizeof(sP_FE2CL_REP_PC_VENDOR_ITEM_BUY_SUCC));
+}
+
+void NPCManager::npcVendorTable(CNSocket* sock, CNPacketData* data) {
+    if (data->size != sizeof(sP_CL2FE_REQ_PC_VENDOR_TABLE_UPDATE))
+        return; // malformed packet
+
+    sP_CL2FE_REQ_PC_VENDOR_TABLE_UPDATE* req = (sP_CL2FE_REQ_PC_VENDOR_TABLE_UPDATE*)data->buf;
+    INITSTRUCT(sP_FE2CL_REP_PC_VENDOR_TABLE_UPDATE_SUCC, resp);
+
+    //TODO: data needs to be read from shopkeeper tabledata
+    //check req->iVendorID and req->iNPC_ID
+    
+    //Exaple Items
+    //shirt
+    //--------------------------------------------------------------
+    sItemBase base1;
+    base1.iID = 60;
+    //??
+    base1.iOpt = 0;
```

Comment:
No need for the `//??`. We know what `iOpt` does, sorta.

Reply:
I left the comment because I wasn't sure how to handle this situation or like what kind of value I should respond because I don't know what it does, should I leave it = 0 so someone can handle it later if its wrong?